tools/common.hpp

```diff
@@ -7,13 +7,21 @@
 
 namespace antler {
 
+   std::string escape_transform(std::string input) {
+      for(auto i=input.begin(); i != input.end(); ++i) {
+         if(*i == '\\' && i+1 != input.end())
+            i = input.erase(i);
+      }
+      return input;
+   }
+

```

Comment:
How about erase-remove idiom?
```suggestion
   std::string escape_transform(std::string input) {
      input.erase(std::remove(input.begin(), input.end(), '\\'), input.end());
      return input;
   }

```

Reply:
erase-remove isn't quite the algorithm we want since it would convert the string "\\-O2" to "-O2" instead of "\-O2".

I added some docs in  52bce66 to address this.

Reply:
I don't think Michal's suggested code does same as original code.
Comment would be useful as I don't quite understand what this function is supposed to do